apps/smart-contracts/core/test/AllowedCallers.test.ts

```diff
@@ -0,0 +1,37 @@
+import chai, { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
+import { smock } from '@defi-wonderland/smock'
+import { allowedCallersFixture } from './fixtures/AllowedCallersFixture'
+import { AllowedCallers } from '../typechain'
+
+chai.use(smock.matchers)
+
+describe('=> AllowedCallers', () => {
+  let restrictedCallers: AllowedCallers
+  let deployer: SignerWithAddress
+  let user: SignerWithAddress
+
+  beforeEach(async () => {
+    ;[deployer, user] = await ethers.getSigners()
+    restrictedCallers = await allowedCallersFixture()
+  })
+
+  describe('initial state', () => {
+    it('has no allowed accounts', async () => {})
+  })
+
+  describe('# setAllowedCallers', () => {
+    it('should revert when callers.length != allowed.length', async () => {})
+
+    it('should set a single account as allowed caller', async () => {})
+
+    it('should set two accounts as allowed', async () => {})
+
+    it('should unset a single account as allowed', async () => {})
+
+    it('should unset one previously allowed account and set one new account', async () => {})
```

Comment:
```suggestion
    it('unsets previously allowed account and sets new account', async () => {})
```
No need for "one" since that is implied by using singular "account"